sw/airborne/modules/relativelocalizationfilter/fmatrix.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) Mario Coppola
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/relativelocalizationfilter/fmatrix.c"
+ * @author Mario Coppola
+ * C matrix functions
+ */
+
+#include "fmatrix.h"
+
+/* Function to add two matrices to eachother */
+void fmat_add(int n_row, int n_col, float* r, float* a, float* b) {
```

Comment:
A few of these functions already exist in pprz_simple_matrix.h. As these are quite generic functions, please augment the existing math functions instead of writing your own private functions. I recommend moving this all to math and using the naming convention used in those functions. I would also recommend changing all the float uses of the matrix_simple macros to your new functions (in a subsequent pull request). compiled functions are generally better than macros for warnings ect.

I know that may seem like a lot and unnecessary for your work but this is the only way that we can improve the code base. If you needed these functions the odds are someone else will need them in the future.